### PR TITLE
Add core commands to Makefile

### PR DESCRIPTION
**Pull request by andrewgy8**
### What's Changed

Adds Makefile commands for MOST docker compose commands.  The client commands were omitted because I could get the client container running.

Addresses #431 

### Technical Description

I would suggest cloning the repo and testing out the commands yourself if you want to test it out.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+.PHONY: env-start env-stop env-restart env-build env-destroy
+.PHONY: migrate upgrade seed lint
+.PHONY: test-users recreate-users upgrade-users migrate-usersrvice
+.PHONY: seed-users lint-users test-events recreate-events
+.PHONY: upgrade-events migrate-events lint-events
+
+PROJECT_NAME := 'my-dev-space'
+DOCKER_COMPOSE_FILE := ./docker-compose-dev.yml
+
+env-start:
```
Comment (apoclyps) [inline Makefile]: `run` would be easier to reason about than `env-start`. 

e.g. `make run`

```suggestion
run:
```

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/make -f
+
+.PHONY: env-start env-stop env-restart env-build env-destroy
```
Comment (apoclyps) [inline Makefile]:
```suggestion
run stop restart build destroy
```

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+.PHONY: seed-users lint-users test-events recreate-events
+.PHONY: upgrade-events migrate-events lint-events
+
+PROJECT_NAME := 'my-dev-space'
+DOCKER_COMPOSE_FILE := ./docker-compose-dev.yml
+
+env-start:
+	docker-compose -p $(PROJECT_NAME) -f $(DOCKER_COMPOSE_FILE) up -d
+
+env-build:
```
Comment (apoclyps) [inline Makefile]:
```suggestion
build:
```

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+	docker-compose -p $(PROJECT_NAME) -f $(DOCKER_COMPOSE_FILE) build --no-cache --pull
+
+env-stop:
```
Comment (apoclyps) [inline Makefile]:
```suggestion
stop:
```

File: Makefile [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+	docker-compose -p $(PROJECT_NAME) -f $(DOCKER_COMPOSE_FILE) down
+
+env-restart:
```
Comment (apoclyps) [inline Makefile]:
```suggestion
restart:
```

File: Makefile [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+	docker-compose -p $(PROJECT_NAME) -f $(DOCKER_COMPOSE_FILE) restart
+
+env-destroy:
```
Comment (apoclyps) [inline Makefile]:
```suggestion
destroy:
```

File: README.md [hunk 7]
```diff
@@ -27,29 +27,25 @@ The following steps will check out the repository, install the dependencies need
 ```bash
 $ git clone https://github.com/apoclyps/my-dev-space
 $ cd my-dev-space
-$ docker-compose -f docker-compose-dev.yml up -d
+$ make env-start
```
Comment (apoclyps) [inline README.md]:
```suggestion
make run
```

File: README.md [hunk 8]
```diff
@@ -60,51 +56,45 @@ To load data into the service for development, the recommended solution is to us
 
 And to tear down the local development stack, simply run:
 
-```
-$ docker-compose -f docker-compose-dev.yml down
+```bash
+$ make env-stop
```
Comment (apoclyps) [inline README.md]:
```suggestion
make stop
```

**Comment by apoclyps** [comment]
@andrewgy8 this is a good first pass; I made a few suggested edits. Ideally, the core commands should be simple and easy to remember: `build`, `run`, `stop`, etc. 

It would be nice to include a list of help fun commands such as `build`, `run`, `stop`, etc.  when you run make as a self-documenting way of providing a description of available commands and what they do.

**Comment by andrewgy8** [comment]
@apoclyps I took your suggestions for removing the `env` prefix. I put it in there to be explicit about what exactly someone is starting to avoid any confusion as to what environment was being targeted, i.e. adding `prod-start`, etc. to make. 

In addition, I added the make self documentation for the basic commands.  Good call.

You'll notice that I stuck with `start` instead of `run`. Run to me, is a bit misleading, as it can be construed for many other operations, `run-tests`, `run-linting`. Run also implies that it is the only command that needs to be executed for someone to start playing around with the services on their local environment.

**Comment by apoclyps** [comment]
@andrewgy8 Interesting idea around the different environments. For the purpose of the Makefile, I would keep it specific to the local development environment and ignore staging/production. Both of these environments come with their own caveats such as requiring elevated permissions for AWS/ Postgres that won't be accessible to most people checking out the codebase.

I'm happy to keep `start`; you're right. `run` can be misleading and it's best to avoid any confusion for new developers 👍

File: README.md [hunk 9]
```diff
@@ -27,29 +27,25 @@ The following steps will check out the repository, install the dependencies need
 ```bash
 $ git clone https://github.com/apoclyps/my-dev-space
 $ cd my-dev-space
-$ docker-compose -f docker-compose-dev.yml up -d
+$ make start
```
Comment (apoclyps) [inline README.md]: It's worth creating a copy of the README for `Setting up a local development environment` with all the docker commands and dropping it into the `docs` directory for prosperity.

**Comment by apoclyps** [comment]
Thanks, @andrewgy8 for this PR; I've had a chance to try it out locally and it looks good. Good job 👍

**Pull request merged**